src/gui/src/tab_devinfo.cpp

```diff
@@ -0,0 +1,352 @@
+
+  pl_rpc_rst = new QLabel(tr("Resets:"), this);
+  pl_rpc_rst->setAlignment(Qt::AlignRight | Qt::AlignVCenter);
+  layout_left->addWidget(pl_rpc_rst, 12, 0);
+  l_rpc_rst = new QLabel(this);
+  l_rpc_rst->setMinimumWidth(120);
+  layout_left->addWidget(l_rpc_rst, 12, 1);
+
+  // Plextor Lifetime
+  hline0 = new QFrame(this);
+  hline0->setFrameStyle(QFrame::HLine | QFrame::Sunken);
+  layout_left->addWidget(hline0, 13, 0, 1, 2);
+
+  pl_life_dn = new QLabel(tr("Discs loaded:"), this);
+  pl_life_dn->setAlignment(Qt::AlignRight | Qt::AlignVCenter);
+  layout_left->addWidget(pl_life_dn, 14, 0);
+  l_life_dn = new QLabel(this);
+  //	l_life_dn->setAlignment(Qt::AlignRight | Qt::AlignVCenter);
+  l_life_dn->setMinimumWidth(120);
+  layout_left->addWidget(l_life_dn, 14, 1);
+
+  pl_life_cr = new QLabel(tr("CD read:"), this);
+  pl_life_cr->setAlignment(Qt::AlignRight | Qt::AlignVCenter);
+  layout_left->addWidget(pl_life_cr, 15, 0);
+  l_life_cr = new QLabel(this);
+  //	l_life_cr->setAlignment(Qt::AlignRight | Qt::AlignVCenter);
+  l_life_cr->setMinimumWidth(120);
+  layout_left->addWidget(l_life_cr, 15, 1);
+
+  pl_life_cw = new QLabel(tr("CD write:"), this);
+  pl_life_cw->setAlignment(Qt::AlignRight | Qt::AlignVCenter);
+  layout_left->addWidget(pl_life_cw, 16, 0);
+  l_life_cw = new QLabel(this);
+  //	l_life_cw->setAlignment(Qt::AlignRight | Qt::AlignVCenter);
+  l_life_cw->setMinimumWidth(120);
+  layout_left->addWidget(l_life_cw, 16, 1);
+
+  pl_life_dr = new QLabel(tr("DVD read:"), this);
+  pl_life_dr->setAlignment(Qt::AlignRight | Qt::AlignVCenter);
+  layout_left->addWidget(pl_life_dr, 17, 0);
+  l_life_dr = new QLabel(this);
+  //	l_life_dr->setAlignment(Qt::AlignRight | Qt::AlignVCenter);
+  l_life_dr->setMinimumWidth(120);
+  layout_left->addWidget(l_life_dr, 17, 1);
+
+  pl_life_dw = new QLabel(tr("DVD write:"), this);
+  pl_life_dw->setAlignment(Qt::AlignRight | Qt::AlignVCenter);
+  layout_left->addWidget(pl_life_dw, 18, 0);
+  l_life_dw = new QLabel(this);
+  //	l_life_dw->setAlignment(Qt::AlignRight | Qt::AlignVCenter);
+  l_life_dw->setMinimumWidth(120);
+  layout_left->addWidget(l_life_dw, 18, 1);
+
+  layout_left->setColumnStretch(0, 2);
+  layout_left->setColumnStretch(1, 3);
+  layout_left->setRowStretch(19, 20);
+
+  int hidx = 0, vidx = 0, hidx0 = 0, rows = 0;
+  MediaCapWidget *cap;
+
+  cap_grid = new QGridLayout;
+  layout->addLayout(cap_grid);
+
+  lc_media = new QLabel("<b>" + tr("Media R/W Capabilities") + "</b>", this);
+  lc_media->setFrameStyle(QFrame::Sunken | QFrame::StyledPanel);
+  lc_media->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Preferred);
+  lc_media->setAlignment(Qt::AlignHCenter | Qt::AlignVCenter);
+  cap_grid->addWidget(lc_media, hidx, 0, 1, CAP_COLS);
+  hidx++;
+  vidx = 0;
+  hidx0 = 0;
+  rows = (sizeof(rw_capabilities) / sizeof(desc64) - 2 + CAP_COLS) / CAP_COLS;
+  for (int idx = 0;
+       rw_capabilities[idx].id || strlen(rw_capabilities[idx].name); idx++) {
```

Comment:
Codacy found an issue: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)